### ISSUE TITLE
what's the difference between rfpimp.importances and rfpimp.permutation_importances?

### ISSUE THREAD

**Issue by hmanz** [parrt/random-forest-importances#16]
I noticed in the README that `rfpimp.importances` was used, whereas [in this blog](https://towardsdatascience.com/explaining-feature-importance-by-example-of-a-random-forest-d9166011959e) they used `from rfpimp import permutation_importances`.

On an unrelated note, I tried both this repo's implementation of permutation importance and also `eli5`'s implementation and got very different results. If anyone has tried both before I would like to hear your experience.

**Comment by parrt** [comment]
I think you can import things as you want.   The permutation importance is should be the same as well as the drop column.  We could have bugs or there could be something different in the way you are constructing trees etc..

**Comment by hmanz** [comment]
@parrt I'm not sure I understand what you are saying. Are you saying that the following three functions in [rfpimp.py](https://github.com/parrt/random-forest-importances/blob/master/src/rfpimp.py) are the same?

1. `importances`
2. `permutation_importances`
3. `dropcol_importances`

If they are the same why are there 3 separate functions? 

In my original question I wanted to know what the difference between 1 and 2 are.

**Comment by parrt** [comment]
I'm simply hoping that the permutation and drop column importance is give you the same answer roughly, but that's not always the case.  Take a look at the associated article.  drop column importance gives you a 0 for highly collinear features.

**Comment by hmanz** [comment]
Okay that makes sense, thank you. The only remaining question I have is what is the difference between the `importances` and `permutation_importances` functions in [rfpimp.py](https://github.com/parrt/random-forest-importances/blob/master/src/rfpimp.py)? 

The sample code in the README has this line: 
`imp = importances(rf, X_test, y_test) # permutation`

The comment is showing that this is running the permutation importance algorithm. So if the `importances` function carries out permutation importance, then what is the purpose of the `permutation_importances` function?

**Comment by parrt** [comment]
Probably just historical bits of code that I eventually merged into importances().
